### PR TITLE
Allow to use all JAXB, Jackson and Swagger annotations

### PR DESCRIPTION
**Pull request by tjuchniewicz**
Fixes gh-424.

### CODE REVIEW

File: src/main/java/com/github/kongchen/swagger/docgen/SwaggerAnnotationIntrospector2.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import io.swagger.jackson.SwaggerAnnotationIntrospector;
+
+/**
+ * Extends SwaggerAnnotationIntrospector with {@link #findRootName(AnnotatedClass)} implementation. See
+ * https://github.com/swagger-api/swagger-core/issues/2104
+ * 
+ * @author Tomasz Juchniewicz
+ *
+ */
+public class SwaggerAnnotationIntrospector2 extends SwaggerAnnotationIntrospector {
```
Comment (who) [inline src/main/java/com/github/kongchen/swagger/docgen/SwaggerAnnotationIntrospector2.java]: This class needs a better name than just adding "2". Please make it descriptive of functionality.

File: src/main/java/com/github/kongchen/swagger/docgen/SwaggerModule2.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.github.kongchen.swagger.docgen;
+
+import com.fasterxml.jackson.databind.module.SimpleModule;
+
+public class SwaggerModule2 extends SimpleModule {
```
Comment (who) [inline src/main/java/com/github/kongchen/swagger/docgen/SwaggerModule2.java]: This class needs a better name than just adding "2". Please make it descriptive of functionality.

File: src/main/java/com/github/kongchen/swagger/docgen/AbstractDocumentSource.java [hunk 3]
```diff
@@ -159,7 +160,17 @@ public void toSwaggerDocuments(String uiDocBasePath, String outputFormats, Strin
...
+            JaxbAnnotationModule jaxbAnnotationModule = new JaxbAnnotationModule();
+            if (apiSource.isUseJAXBAnnotationProcessorAsPrimary()) {
+                jaxbAnnotationModule.setPriority(Priority.PRIMARY);    
+            } else {
+                jaxbAnnotationModule.setPriority(Priority.SECONDARY);
+            }
+            objectMapper.registerModule(jaxbAnnotationModule);
+            
+            // to support @ApiModel on class level.
+            // must be registered only if we use JaxbAnnotationModule before. Why?
```
Comment (who) [inline src/main/java/com/github/kongchen/swagger/docgen/AbstractDocumentSource.java]: This comment is confusing. Could you please clarify?

Comment (tjuchniewicz) [inline src/main/java/com/github/kongchen/swagger/docgen/AbstractDocumentSource.java]: I do not understand plugin or swagger-core internals well. 

1. I don't understand why this happens but when JAXB is disabled Swagger annotations are used. This is fine. But when I register JAXB module (as PRIMARY or SECONDARY) Swagger annotations are not used anymore. I had to register Swagger explicitly.
2. When I manually register default SwaggerModule `@ApiModel` is ignored. That's why I use enhanced SwaggerModule. But it works fine if JAXB module is not loaded. 

Confusing...

File: src/main/java/com/github/kongchen/swagger/docgen/SwaggerAnnotationIntrospector2.java (hunk 1 above)
Comment (tjuchniewicz) [inline src/main/java/com/github/kongchen/swagger/docgen/SwaggerAnnotationIntrospector2.java]: Renamed to `EnhancedSwaggerAnnotationIntrospector`

File: src/main/java/com/github/kongchen/swagger/docgen/SwaggerModule2.java (hunk 2 above)
Comment (tjuchniewicz) [inline src/main/java/com/github/kongchen/swagger/docgen/SwaggerModule2.java]: Renamed to `EnhancedSwaggerModule`

**Pull request merged**